### ISSUE TITLE
WASM_X64: `textual` `64-bit` assembly generated segment faults

### ISSUE THREAD

**Issue by Shaikh-Ubaid** [lcompilers/lpython#1353]
The `textual` `64-bit` assembly format that is currently generated seems to `segment fault`. I think we can fix in some other `PR`.

_Originally posted by @Shaikh-Ubaid in https://github.com/lcompilers/lpython/issues/1338#issuecomment-1334710796_

**Comment by Shaikh-Ubaid** [comment]
The reason it fails is because in the `asm` text format we generate, we fix (or equate) the start location of instructions based on the binary code that we generate. When using `nasm` to compute the `asm` text code, it happens that `nasm` converts the instructions into binary code using a combination of `64-bit` and/or `32-bit` instructions. It seems to use an instruction size (`64-bit` or `32-bit`) depending on the instruction. Since, some instruction(s) in the `asm` text code can be encoded using `32-bit` size instruction (and `32-bit` sized operands), if such an instruction(s) are before the `_start` label, it leads to an (undesirable) offset in the start position (which we had already fixed/emitted in the text format). Thus, the start position we emitted becomes incorrect.
